datafusion/core/tests/sql/functions.rs

```diff
@@ -357,3 +357,240 @@ async fn coalesce_mul_with_default_value() -> Result<()> {
     assert_batches_eq!(expected, &actual);
     Ok(())
 }
+
+#[tokio::test]
+async fn count_basic() -> Result<()> {
```

Comment:
As `COUNT` and `AVG` are aggregate functions, these tests probably belong in aggregates.rs or something similar -- we can move them as a follow on PR

Reply:
(I am working on this)

Reply:
https://github.com/apache/arrow-datafusion/pull/2333#